cmd/start.go

```diff
@@ -32,48 +33,53 @@ func startCmd() *cobra.Command {
 			}
 			defer m.Close()
 
-			file, err := os.Open(fileName)
-			if err != nil {
-				return fmt.Errorf("opening migration file: %w", err)
-			}
-			defer file.Close()
+			return runMigrationFromFile(cmd.Context(), m, fileName, complete)
+		},
+	}
 
-			migration, err := migrations.ReadMigration(file)
-			if err != nil {
-				return fmt.Errorf("reading migration file: %w", err)
-			}
+	startCmd.Flags().BoolVarP(&complete, "complete", "c", false, "Mark the migration as complete")
 
-			sp, _ := pterm.DefaultSpinner.WithText("Starting migration...").Start()
-			cb := func(n int64) {
-				sp.UpdateText(fmt.Sprintf("%d records complete...", n))
-			}
+	return startCmd
+}
 
-			err = m.Start(cmd.Context(), migration, cb)
-			if err != nil {
-				sp.Fail(fmt.Sprintf("Failed to start migration: %s", err))
-				return err
-			}
+func runMigrationFromFile(ctx context.Context, m *roll.Roll, fileName string, complete bool) error {
+	file, err := os.Open(fileName)
+	if err != nil {
+		return fmt.Errorf("opening migration file: %w", err)
+	}
```

Comment:
Now that this is a function I think that `file.Close()` can be deferred here rather than called later.